3 - Editores (VIM, EMACS, VSCode)/vim.md

```diff
@@ -0,0 +1,92 @@
+Adicionalmente, devido a sua origem, o editor força o usuário a nunca tirar a mão de perto das letras, e isso permite que ações sejam encadeadas sem interrupções. 
+Além disso, vim permite que funções novas de edição sejam criadas, dando liberdade ao usuário de customizar, através do arquivo .vimrc.
+
+# Os básicos.
+
+## Como abrir um arquivo:
+Para abrir um arquivo com vim, basta digitarmos no terminal: `vim [nome-do-arquivo]`. Caso o arquivo não exista, o editor tratará de criar esse arquivo para você.
+
+## Ué, porque o que eu escrevo não aparece na tela?
+
+Se você acabou de abrir o editor e já tentou escrever alguma coisa, vai perceber que o resultado não foi nada esperado. 
+Isso se dá porque vim funciona através de modos: Quando abrimos o editor, ele está no modo normal, que permite a navegação pelo texto, mas não a edição.
+Para editarmos, basta digitar `i` dentro do modo normal, o que nos levará ao modo de inserção, nele podemos digitar sem obstáculos. 
+Para sairmos do modo de inserção, basta apertar a tecla Esc que voltaremos ao modo normal. Essa lógica vale para todos os modos que você se encontrar.
+
+## Ok, mas como que eu salvo um arquivo?
+Você não sai. MUAHAHAHAHAHAH. Brincadeira. Para salvarmos um arquivo, precisamos dizer ao editor que vamos executar um comando.
+Para isso, precisamos estar no modo normal e então digitar `:w` e apertar a tecl Enter. Os dois pontos ativam o modo de linha de comando, 
+já o caractere `w` indica o comando de salvar o conteúdo escrito na tela para o arquivo. 
+
+A partir de agora é só colinha para refrescar os comandos:
+
+# Modo Normal
+## Movimentação
+
+-  Toda movimentação pode receber uma quantidade antes que indica quantas vezes será repetida. Ex: `4j` ou `50l`
+-  Por caracteres: `h`(esquerda), `j`(baixo), `k`(cima), `l`(direita)
+-  Por "palavra": `w`(começo da próxima palavra, `e`(final da próxima palavra), `b` (começo da palavra anterior)
+-  Por linha (não aceita quantidade): `0`(começo da linha), `$`(final da linha)
+-  Por arquivo: `gg`(Começo do arquivo), `G`(Final do arquivo) 
+
+## Desfazer ações
+-  Desfazer: `u`
+-  Desfazer o desfazer: `R`
+
+## Deletar
+-  Aceita quantidades antes da ação
+-  Deletar por movimento: `d` + movimento
+-  Deletar linha inteira: `dd`
+-  Deletar até o final da linha: `D`
+
+## Copiar e Colar
+-  Copiar por movimento: `y` + movimento
+-  Copiar linha inteira: `Y` ou `yy`
+-  Colar antes do cursor: `P`
+-  Colar depois do cursor: `p`
+
+# Modo de Inserção (entra-se a partir do modo normal)
+-  Inserir antes do cursor: `i`
+-  Adicionar ao final do cursor: `a` 
+-  Sair do modo normal : Tecla Esc
+
+# Modo visual (entra-se a partir do modo normal)
+-  Funcionalidade igual ao modo normal, porém selecionando texto.
+-  Ativar modo visual por caractere: `v`
+-  Ativar modo visual por linha: `V`
+-  Ativar modo visual por coluna: `CTRL+V`
+
+
+# Modo de comandos (entra-se a partir do modo normal ou visual)
```

Comment:
Coloca a opcao `:%s/adsf/asdf/` tambem. O `%` indica "no arquivo inteiro" em vez de "nessa linha"

Reply:
Então, isso eu quis omitir por causa do jeito que eu ia ensinar, que é fazendo a seleção pelo modo visual e então fazendo o search and replace. Mas acho melhor colocar também